presto-spi/src/test/java/com/facebook/presto/spi/block/BenchmarkFloorMod.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.block;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+@SuppressWarnings("MethodMayBeStatic")
+@State(Scope.Thread)
+@OutputTimeUnit(MILLISECONDS)
+@Fork(3)
+@Warmup(iterations = 10, time = 500, timeUnit = MILLISECONDS)
+@Measurement(iterations = 10, time = 500, timeUnit = MILLISECONDS)
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkFloorMod
+{
+    static final int numHashCodes = 1000000;
+    int hashTableSize = ThreadLocalRandom.current().nextInt() & 0x7fff_ffff;
+
+    @Benchmark
+    public void ReduceHashPosition1(RandBenchmarkData data, Blackhole blackhole)
+            throws Throwable
+    {
+        long[] hashcodes = data.getHashcodes();
+
+        for (long i : hashcodes) {
```

Comment:
Why are we looping instead of doing it the way I explained during our face to face conversation? The computation is insensitive to the values of the hashes, so there's no benefit in using an array of hashcodes vs a single `long` variable in the state.

Reply:
@martint I understand. @dain had question about if using a single value would affect the test result, plus in this looping way I can check if the result is correct and verify the distribution easily if I want, so I changed it to the looping way. In both ways the performance results match.